controllers/openstackbaremetalset_controller.go

```diff
@@ -788,7 +788,11 @@ func (r *OpenStackBaremetalSetReconciler) ensureBaremetalHosts(
 	newBmhsNeededCount := instance.Spec.Count - len(existingBaremetalHosts.Items)
 
 	// Sort the list of available BaremetalHosts
-	sort.Strings(availableBaremetalHosts)
+	sort.Slice(availableBaremetalHosts[:], func(i, j int) bool {
+		// strings.Compare(a, b) returns -1 when a < b, which is the comparison we want
+		// for ordering alphabetically as sort.Slice expects a "less" function
+		return strings.Compare(availableBaremetalHosts[i].Name, availableBaremetalHosts[j].Name) == -1
```

Comment:
Expect you could just use <, that what's used for StringSlice https://cs.opensource.google/go/go/+/refs/tags/go1.21.5:src/sort/sort.go;l=154

Reply:
Ack, I will try that instead to keep things simpler